evalml/pipelines/utils.py

```diff
@@ -105,10 +109,8 @@ def _get_preprocessing_components(
         text_columns
     ):
         pp_components.append(Imputer)
-    index_and_unknown_columns = list(
-        X.ww.select(["index", "unknown"], return_schema=True).columns
-    )
-    if len(index_and_unknown_columns) > 0:
+
+    if X.ww.index is not None:
```

Comment:
So if the pipeline has an index and unknown columns, it will have two drop columns transformers? I think that'll work but it'll be confusing because AutoMLSearch will only update the parameters for the first drop columns transformer.

Reply:
So the second one won't "do" anything I believe

Reply:
Agreed, I don't think we should have 2 of the same components if we don't need/use both

Reply:
Agreed--the code in AutoML search here means that we should be covered without needing two DropColumn components, right?

This makes me question if there was a way to pass component instances instead of component classes along, s.t. we could have a DropColumn(index) and DropColumn(unknown_cols) component passed to AutoML rather than having automl handle the parameters to initialize separately. But OOS for this discussion I guess 🤷‍♀️ 


```
            index_and_unknown_columns = list(
                self.X_train.ww.select(["index", "unknown"], return_schema=True).columns
            )
            unknown_columns = list(
                self.X_train.ww.select("unknown", return_schema=True).columns
            )
            index_and_unknown_columns = _put_into_original_order(
                self.X_train, index_and_unknown_columns
            )
            if len(index_and_unknown_columns) > 0 and drop_columns is None:
                parameters["Drop Columns Transformer"] = {
                    "columns": index_and_unknown_columns
                }
                if len(unknown_columns):
                    logger.info(
                        f"Removing columns {unknown_columns} because they are of 'Unknown' type"
 ```

Reply:
Should be good to go now.  I think I misunderstood what was happening!